daisy/workflow/validate.go

```diff
@@ -141,6 +142,9 @@ func (w *Workflow) validateRequiredFields() error {
 	if w.Name == "" {
 		return errors.New("must provide workflow field 'Name'")
 	}
+	if !rfc1035Rgx.MatchString(strings.ToLower(w.Name)) {
```

Comment:
So I tried a sanitize() function and saw some potential issues. It works fine for GCE resources but gets funky for GCS resources. do we just sanitize everything even though GCS can take the special characters? Do we not sanitize GCS paths and just live with potential white space?
Also what do we do about needing a letter as the first character, print out any numbers? ASlow what about the special characters? 


func sanitize(n string) string {
	noSpaces := strings.Replace(n, " ", "-", -1)
	return strings.ToLower(strings.Replace(noSpaces, "_", "-", -1))
}